meteor/client/ui/SegmentTimeline/SourceLayerItem.tsx

```diff
@@ -102,6 +103,13 @@ export const SourceLayerItem = withTranslation()(
 			return Math.round(this.props.timeScale * time)
 		}
 
+		shouldSwapDurationLabel = (): boolean => {
+			if (this.props.part && this.props.partStartsAt !== undefined && !this.props.isLiveLine) {
+				return this.state.leftAnchoredWidth > this.state.elementWidth - 10
```

Comment:
Where does the 10 come from?

Reply:
An arbitrary value making it swap at the right point